packages/core/src/test-utils/mock-tool.ts

```diff
@@ -37,6 +46,24 @@ interface MockToolOptions {
   params?: object;
 }
 
+export interface MockModifiableToolShouldConfirmExecuteParams {
+  filePath: string;
+  currentContent: string;
+  proposedContent: string;
+}
+
+// disallow shouldConfirmExecute in favor of passing shouldConfirmExecuteParams
```

Comment:
Consider adjusting this comment to be a docblock and rather than it being a note, prefer to describe this type (e.g. "Some tool options that excludes xyz because of abc").